### PR TITLE
Add common AMQP connection provider + Make SAS generation more generic

### PR DESCRIPTION
**Pull request by jseely**

### CODE REVIEW

File: provider/provider.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	p.Logger.SetLevel(log.WarnLevel)
+	return p
+}
+
+// NewConnection generates a new AMQP connection
+func (p *Provider) NewConnection() (*amqp.Client, error) {
+	return amqp.Dial(p.GetAmqpHostURI(),
+		amqp.ConnSASLAnonymous(),
+		amqp.ConnMaxSessions(65535),
+		amqp.ConnProperty("product", "MSGolangClient"),
```
Comment (devigned) [inline provider/provider.go]: These options can't be added here and would need to be provided from the caller. I'd recommend adding them as variadic options on `NewConnection`

File: provider/provider.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+//	SOFTWARE
+
+import (
+	"context"
+	"runtime"
+
+	"github.com/Azure/azure-amqp-common-go/auth"
+	"github.com/Azure/azure-amqp-common-go/cbs"
+
+	log "github.com/sirupsen/logrus"
```
Comment (devigned) [inline provider/provider.go]: Please remove the logger reference.

**Comment by devigned** [comment]
I've opened #17 to make SAS more generic. I'm going to close this as it's getting stale. If you'd like to pursue this PR further after rebasing changes, please reopen.

**Comment by jseely** [comment]
Thanks for doing that, busy month and forgot I still had this open. I'll take a look at #17 and see if that covers everything I need.

**Pull request closed without merging**